store/sqlstore/scheme_store.go

```diff
@@ -293,66 +305,105 @@ func (s *SqlSchemeStore) Delete(schemeId string) (*model.Scheme, error) {
 	s.Channel().ClearCaches()
 
 	// Delete the roles belonging to the scheme.
+
+	inQuery := sq.Or{}
```

Comment:
You don't need an Or and then append `sq.Eq` repeatedly to it. `sq.Eq` already supports an array. So what we need to do is to keep this part of the code which creates the `roleNames` slice as is. And just do `sq.Eq{"Name": roleNames}` in the `Where` clause.

Reply:
That is great. You are right, then we don't need to reassign anything. Corrected this.